test/general/TestReachability.py

```diff
@@ -67,3 +68,18 @@ def testDefaultEmptyStateCanReachSomething(self):
                             locations.add(location)
                     self.assertGreater(len(locations), 0,
                                        msg="Need to be able to reach at least one location to get started.")
+
+    def testEarlyItemsFilterDoesNotCrash(self) -> None:
+        """
+        This makes sure `filter_early_locations` is run in every world (within `distribute_early_items`).
```

Comment:
> If the objective is really to "test that filter_early_locations is properly implemented" (i.e., the system under test is an individual world), I would (for each world type) create a solo multiworld, let it generate till after prefill, and then explictily call filter_early_locations. Then you could run some test assertions on the return values. You could do this twice: passing zero locations and passing all of that world's unfilled locations.

This doesn't test it with a realistic set of locations.
If the world developer is trying to fix a problem with that test failing, it's more likely to not fix the problem for a more realistic case.
A way to get a more realistic set of locations would be to duplicate the code in `distribute_early_items`, but then, if the code in `distribute_early_items` changes, the test becomes less relevant or requires more maintenance.
I also don't like relying on a concrete world (Clique) in the general tests. Duplicating the code in `distribute_early_items` would also help avoid that.

Do you not think there's any value in testing each world's `filter_early_locations` in the context of `distribute_early_items`?

The 2 tests you've suggested would be valuable additions, But neither of them covers the objective that I think would be most valuable: testing each world's `filter_early_locations` in the context of `distribute_early_items`.

I think the test would be a lot more valuable if it calls `distribute_early_items` in a state that ensures that `filter_early_locations` will be called.

I started going down the road of using a generic world instead of Clique and soon realized it would be a lot more work than I first anticipated. That's when I switched to using Clique.